keras/layers/rnn/bidirectional.py

```diff
@@ -187,7 +187,7 @@ def compute_output_shape(self, sequences_shape, initial_state_shape=None):
         if self.return_state:
             if self.merge_mode is None:
                 return output_shape + state_shape + copy.copy(state_shape)
-            return [output_shape] + state_shape + copy.copy(state_shape)
+            return [output_shape] + [state_shape] + [copy.copy(state_shape)]
```

Comment:
This line should be 

```
return tuple(output_shape) + tuple(state_shape) + tuple(state_shape)
```

Reply:
Ok.I thought it is rendered as list. Also please check the line no.185

https://github.com/keras-team/keras/blob/d364c567a98a3fc0915cbbf9ea6de428f695cf17/keras/layers/rnn/bidirectional.py#L184-L185

Is that also needs to be a tuple ?

Reply:
My mistake. It should be `return (output_shape, state_shape, copy.copy(state_shape))`. I think the line above also needs to be modified. Is this properly tested?

Reply:
> My mistake. It should be `return (output_shape, state_shape, copy.copy(state_shape))`. I think the line above also needs to be modified. Is this properly tested?

With this correction Keras3 Outputs a nested tuple:
```
(<KerasTensor shape=(None, 512), dtype=float32, sparse=False, name=keras_tensor_114>,
 (<KerasTensor shape=(None, 256), dtype=float32, sparse=False, name=keras_tensor_115>,
  <KerasTensor shape=(None, 256), dtype=float32, sparse=False, name=keras_tensor_116>),
 (<KerasTensor shape=(None, 256), dtype=float32, sparse=False, name=keras_tensor_117>,
  <KerasTensor shape=(None, 256), dtype=float32, sparse=False, name=keras_tensor_118>))
```

Keras 2 outputs a List(No nesting):

```
[<KerasTensor: shape=(None, 512) dtype=float32 (created by layer 'bidirectional')>,
 <KerasTensor: shape=(None, 256) dtype=float32 (created by layer 'bidirectional')>,
 <KerasTensor: shape=(None, 256) dtype=float32 (created by layer 'bidirectional')>,
 <KerasTensor: shape=(None, 256) dtype=float32 (created by layer 'bidirectional')>,
 <KerasTensor: shape=(None, 256) dtype=float32 (created by layer 'bidirectional')>]
```
Please may refer attached [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/d6fd8a1b6f4e56b08c321dcf73a6a065/19328_r2.ipynb).

I am not sure which is intended and more compatible.

Reply:
Looks like the first thing to do is add an appropriate unit test. Then the version that works will be the right one.

Reply:
The proposal need changes.And also will comeup with testcase.